datafusion/physical-expr/src/aggregate/groups_accumulator/accumulate.rs

```diff
@@ -139,10 +139,14 @@ impl NullState {
             // no nulls, no filter,
             (false, None) => {
                 let iter = group_indices.iter().zip(data.iter());
+
                 for (&group_index, &new_value) in iter {
-                    seen_values.set_bit(group_index, true);
                     value_fn(group_index, new_value);
                 }
+                // update seen values in separate loop
```

Comment:
the idea is to use a separate loop

Note we can skip this update entirely once all the groups have been seen -- I will explore if I can figure out how to do that next.